public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,347 @@
+import { ASSOCIATED_DETECTOR_ACTION } from '../utils/constants';
+import { ConfirmUnlinkDetectorModal } from '../components/ConfirmUnlinkDetectorModal/ConfirmUnlinkDetectorModal';
+
+interface ConfirmModalState {
+  isOpen: boolean;
+  action: ASSOCIATED_DETECTOR_ACTION;
+  isListLoading: boolean;
+  isRequestingToClose: boolean;
+  affectedDetector: DetectorListItem;
+}
+
+function AssociatedDetectors({ embeddable, closeFlyout, core, setMode }) {
+  const dispatch = useDispatch();
+  const allDetectors = useSelector((state: AppState) => state.ad.detectorList);
+  const isRequestingFromES = useSelector(
+    (state: AppState) => state.ad.requesting
+  );
+  const [isLoadingFinalDetectors, setIsLoadingFinalDetectors] =
+    useState<boolean>(true);
+  const isLoading = isRequestingFromES || isLoadingFinalDetectors;
+  const errorGettingDetectors = useSelector(
+    (state: AppState) => state.ad.errorMessage
+  );
+  const embeddableTitle = embeddable.getTitle();
+  const [selectedDetectors, setSelectedDetectors] = useState(
+    [] as DetectorListItem[]
+  );
+
+  const [detectorToUnlink, setDetectorToUnlink] = useState(
+    {} as DetectorListItem
+  );
+  const [confirmModalState, setConfirmModalState] = useState<ConfirmModalState>(
+    {
+      isOpen: false,
+      //@ts-ignore
+      action: null,
+      isListLoading: false,
+      isRequestingToClose: false,
+      affectedDetector: {} as DetectorListItem,
+    }
+  );
+
+  // Establish savedObjectLoader for all operations on vis_augment saved objects
+  const savedObjectLoader: SavedObjectLoader = getSavedFeatureAnywhereLoader();
+
+  useEffect(() => {
+    if (
+      errorGettingDetectors &&
+      !errorGettingDetectors.includes(SINGLE_DETECTOR_NOT_FOUND_MSG)
+    ) {
+      console.error(errorGettingDetectors);
+      core.notifications.toasts.addDanger(
+        typeof errorGettingDetectors === 'string' &&
+          errorGettingDetectors.includes(NO_PERMISSIONS_KEY_WORD)
+          ? prettifyErrorMessage(errorGettingDetectors)
+          : 'Unable to get all detectors'
+      );
+      setIsLoadingFinalDetectors(false);
+    }
+  }, [errorGettingDetectors]);
+
+  // Update modal state if user decides to close modal
+  useEffect(() => {
+    if (confirmModalState.isRequestingToClose) {
+      if (isLoading) {
+        setConfirmModalState({
+          ...confirmModalState,
+          isListLoading: true,
+        });
+      } else {
+        setConfirmModalState({
+          ...confirmModalState,
+          isOpen: false,
+          isListLoading: false,
+          isRequestingToClose: false,
+        });
+      }
+    }
+  }, [confirmModalState.isRequestingToClose, isLoading]);
+
+  useEffect(() => {
+    getDetectors();
+  }, []);
+
+  // Handles all changes in the assoicated detectors such as unlinking or new detectors associated
+  useEffect(() => {
+    // Gets all augmented saved objects
+    savedObjectLoader
+      .findAll()
```

Comment:
You should be able to pass some arguments to `findAll()` to handle filtering for you ([ref](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/feature/feature-anywhere/src/plugins/saved_objects/public/saved_object/saved_object_loader.ts#L131)). For example, passing in search fields on `originPlugin` and/or `visId` with the values you want. This way you wouldn't need to have all of the helper fns.

Reply:
Would you prefer that this is used then what I already have coded, just in terms I know this already works and has been tested. We can also have any future optimization or changes in our control here.

Reply:
I would prefer the way I recommended since it would allow us to leverage OpenSearch and be more optimized and simpler. Also, it would be less code we have to maintain or test.

Reply:
Discussed offline, filtering through FindAll is limited to title and description, issue is here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3566